Source/AFResult.swift

```diff
@@ -0,0 +1,247 @@
+//
+//  AFResult.swift
+//
+//  Copyright (c) 2014-2019 Alamofire Software Foundation (http://alamofire.org/)
+//
+//  Permission is hereby granted, free of charge, to any person obtaining a copy
+//  of this software and associated documentation files (the "Software"), to deal
+//  in the Software without restriction, including without limitation the rights
+//  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+//  copies of the Software, and to permit persons to whom the Software is
+//  furnished to do so, subject to the following conditions:
+//
+//  The above copyright notice and this permission notice shall be included in
+//  all copies or substantial portions of the Software.
+//
+//  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+//
+
+import Foundation
+
+public typealias AFResult<T> = Result<T, Error>
+
+// MARK: - CustomStringConvertible
+
+extension AFResult: CustomStringConvertible {
+    /// The textual representation used when written to an output stream, which includes whether the result was a
+    /// success or failure.
+    public var description: String {
+        switch self {
+        case .success:
+            return "SUCCESS"
+        case .failure:
+            return "FAILURE"
+        }
+    }
+}
+
+// MARK: - CustomDebugStringConvertible
+
+extension AFResult: CustomDebugStringConvertible {
+    /// The debug textual representation used when written to an output stream, which includes whether the result was a
+    /// success or failure in addition to the value or error.
+    public var debugDescription: String {
+        switch self {
+        case .success(let value):
+            return "SUCCESS: \(value)"
+        case .failure(let error):
+            return "FAILURE: \(error)"
+        }
+    }
+}
```

Comment:
We'll probably want to do this a different way using internal APIs. Otherwise we are really polluting the Result behavior publicly. Not something that you need to solve in this PR @AtomicCat.

Reply:
We shouldn't override these at all, as it was decided the compiler generated strings are good enough when `Result` was added to the standard library.

Reply:
The `debugDescription`s of the Response types rely on these extensions. The Swift Result doesn't appear to conform to either of the string convertible protocols.

Reply:
Right. Rather than accessing the property directly, we can use string interpolation to get the description of the value. That would match what users expect for the type’s description.

Reply:
Here's one way that could look:

```swift
extension String.StringInterpolation {
    mutating func appendInterpolation(afResult value: AFResult, verbose: Bool = false) {
        if verbose {
            switch self {
	        case .success(let value): return "SUCCESS: \(value)"
	        case .failure(let error): return "FAILURE: \(error)"
	    }
        } else {
            switch self {
	        case .success: return "SUCCESS"
	        case .failure: return "FAILURE"
	    }
        }
    }
}
```

Reply:
Sorry, I didn't mean we should do anything custom here, just our current uses of `result.description` or `result.debugDescription` could just be replaced by `"\(result)". Unless there's an important bit of data we want to provide, but I don't think there is.